platform-configure.sh

```diff
@@ -15,120 +15,78 @@ set -e
+
+    PROTONET_PUBKEY="-----BEGIN PUBLIC KEY-----
+    MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA5CfJQVP2yJlcMu/3/RxD
+    KnOvcxD40VWsDiUn/FDXlcgQWpg/xH2a7LD9bpD4c3+jWtUst+I7ZhL11YiyfQDr
+    Afw9m11RiHtl+fvJfLg8PwuQ25jc5Cf/hLn+NpnFxL4vlifNWljIoIh17j3KE0hj
+    jd/V7435gkIm0eIvTiebn4cposzh74XrlOnsGyTTyPJ4IMcnS3zYdOIAeTKoSMea
+    rUIsXC8jYMQtua8q96eqM3bPsvFLBWRRQoTfRtVSfydNbZp+i1SixVKo4oDz9UmF
+    fNLAJRPgRI+pXV0O6MdmPtKu5dQNkVGAYm7RWbxZctGxsOArXE43OjqE6kGLVabw
+    7QIDAQAB
+    -----END PUBLIC KEY-----"
 }
 
-function is_update_key_protonet() {
-	key_path="/usr/share/update_engine/update-payload-key.pub.pem"
-  current_digest=$(cat "$key_path" | /usr/bin/sha1sum | cut -f1 -d ' ')
-	protonet_digest=$(echo "$PROTONET_PUBKEY" | /usr/bin/sha1sum | cut -f1 -d ' ')
-  if [[ "$current_digest" == "$protonet_digest" ]]; then
-    return 0
-  else
-    return 1
-  fi
+
+function set_status() {
+    mkdir -p /etc/protonet/system
+    echo "$@" > /etc/protonet/system/configure-script-status
 }
 
-function enable_protonet_updates() {
-  if [[ -z "${PLATFORM_SYS_GROUP}" ]]; then
-    if [ -e ${UPDATE_ENGINE_CONFIG} ]; then
-      PLATFORM_SYS_GROUP=$(cat ${UPDATE_ENGINE_CONFIG} | grep '^GROUP=' | cut -f2 -d '=')
-      echo "Using OS group '${PLATFORM_SYS_GROUP}' from ${UPDATE_ENGINE_CONFIG}."
+function is_update_key_protonet() {
+    key_path="/usr/share/update_engine/update-payload-key.pub.pem"
+    current_digest=$(cat "$key_path" | /usr/bin/sha1sum | cut -f1 -d ' ')
+    protonet_digest=$(echo "$PROTONET_PUBKEY" | /usr/bin/sha1sum | cut -f1 -d ' ')
+    if [[ "$current_digest" == "$protonet_digest" ]]; then
+        return 0
     else
-      PLATFORM_SYS_GROUP="protonet"
-      echo "No OS group given. Using '${PLATFORM_SYS_GROUP}' (default group)."
+        return 1
     fi
-  else
-    echo "Using OS group '${PLATFORM_SYS_GROUP}' from the command line."
-  fi
+}
 
+function prepare_os_update() {
 	# in case there was an automatic update already running
 	update_engine_client -reset_status
 
 	# just in case someone left a key mount
-  umount /usr/share/update_engine/update-payload-key.pub.pem &>/dev/null || true
-
+    umount /usr/share/update_engine/update-payload-key.pub.pem &>/dev/null || true
 	if ! is_update_key_protonet; then
-    echo "$PROTONET_PUBKEY" > /tmp/protonet-image.pub.pem
-    mount --bind /tmp/protonet-image.pub.pem /usr/share/update_engine/update-payload-key.pub.pem
+        echo "$PROTONET_PUBKEY" > /tmp/protonet-image.pub.pem
+        mount --bind /tmp/protonet-image.pub.pem /usr/share/update_engine/update-payload-key.pub.pem
   fi
-
-	# reset backoff timestamp
-  rm -f /var/lib/update_engine/prefs/backoff-expiry-time
-
-	# configure update source
-  echo | tee /etc/coreos/update.conf &>/dev/null <<- EOM
-GROUP=$PLATFORM_SYS_GROUP
-SERVER=https://coreos-update.protorz.net/update
-REBOOT_STRATEGY=off
-EOM
-
-	# apply changes to update-engine
-	systemctl restart update-engine.service
```

Comment:
There is no telling how long it will take to download the platform configure image. Update Engine needs to be restarted immediately after bind-mounting the new key.